src/modules/ekf2/EKF/aid_sources/range_finder/range_finder_consistency_check.cpp

```diff
@@ -36,56 +36,146 @@
  */
 
 #include <aid_sources/range_finder/range_finder_consistency_check.hpp>
+#include "ekf_derivation/generated/range_validation_filter_h.h"
 
-void RangeFinderConsistencyCheck::update(float dist_bottom, float dist_bottom_var, float vz, float vz_var,
-		bool horizontal_motion, uint64_t time_us)
+using namespace matrix;
+
+void RangeFinderConsistencyCheck::init(const float &z, const float &z_var, const float &dist_bottom,
+				       const float &dist_bottom_var)
 {
-	if (horizontal_motion) {
-		_time_last_horizontal_motion = time_us;
-	}
+	// assume no correlation between z and dist_bottom
+	// values defined with symforce for H[1, -1] , P = [z_var, 0; 0, z_var + dist_bottom_var], K = [0,1]
+	_P(RangeFilter::z.idx, RangeFilter::z.idx) = z_var;
+	_P(RangeFilter::z.idx, RangeFilter::terrain.idx) = -z_var;
+	_P(RangeFilter::terrain.idx, RangeFilter::z.idx) = -z_var;
+	_P(RangeFilter::terrain.idx, RangeFilter::terrain.idx) = 5 * (dist_bottom_var + z_var);
```

Comment:
Don't you find this result a bit unexpected? Negative correlation between height and terrain height?